platforms/ios/src/TangramMap/TGMapViewController.mm

```diff
@@ -535,56 +535,135 @@ - (void)setCameraType:(TGCameraType)cameraType
+    _shoveGestureRecognizer = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(respondToShoveGesture:)];
+    _shoveGestureRecognizer.minimumNumberOfTouches = 2;
+    _longPressGestureRecognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(respondToLongPressGesture:)];
 
-    //5. Rotate
-    UIRotationGestureRecognizer* rotationRecognizer = [[UIRotationGestureRecognizer alloc]
-                                                       initWithTarget:self action:@selector(respondToRotationGesture:)];
+    // Use the delegate method 'shouldRecognizeSimultaneouslyWithGestureRecognizer' for gestures that can be concurrent
+    _panGestureRecognizer.delegate = self;
+    _pinchGestureRecognizer.delegate = self;
+    _rotationGestureRecognizer.delegate = self;
 
-    //6. Shove
-    UIPanGestureRecognizer* shoveRecognizer = [[UIPanGestureRecognizer alloc]
-                                               initWithTarget:self action:@selector(respondToShoveGesture:)];
-    shoveRecognizer.minimumNumberOfTouches = 2;
+    [self.view addGestureRecognizer:_tapGestureRecognizer];
+    [self.view addGestureRecognizer:_doubleTapGestureRecognizer];
+    [self.view addGestureRecognizer:_panGestureRecognizer];
+    [self.view addGestureRecognizer:_pinchGestureRecognizer];
+    [self.view addGestureRecognizer:_rotationGestureRecognizer];
+    [self.view addGestureRecognizer:_shoveGestureRecognizer];
+    [self.view addGestureRecognizer:_longPressGestureRecognizer];
+}
 
-    //7. Long press
-    UILongPressGestureRecognizer* longPressRecognizer = [[UILongPressGestureRecognizer alloc]
-                                                         initWithTarget:self action:@selector(respondToLongPressGesture:)];
+- (void) setTapGestureRecognizer:(UITapGestureRecognizer *)recognizer
+{
+    if (_tapGestureRecognizer) {
+        [self.view removeGestureRecognizer:_tapGestureRecognizer];
+    }
+    _tapGestureRecognizer = recognizer;
+    [self.view addGestureRecognizer:_tapGestureRecognizer];
+}
 
-    // Use the delegate method 'shouldRecognizeSimultaneouslyWithGestureRecognizer' for gestures that can be concurrent
-    panRecognizer.delegate = self;
-    pinchRecognizer.delegate = self;
-    rotationRecognizer.delegate = self;
-
-    /* Setup gesture recognizers */
-    [self.view addGestureRecognizer:tapRecognizer];
-    [self.view addGestureRecognizer:doubleTapRecognizer];
-    [self.view addGestureRecognizer:panRecognizer];
-    [self.view addGestureRecognizer:pinchRecognizer];
-    [self.view addGestureRecognizer:rotationRecognizer];
-    [self.view addGestureRecognizer:shoveRecognizer];
-    [self.view addGestureRecognizer:longPressRecognizer];
+- (void) setDoubleTapGestureRecognizer:(UITapGestureRecognizer *)recognizer
+{
+    if (_doubleTapGestureRecognizer) {
+        [self.view removeGestureRecognizer:_doubleTapGestureRecognizer];
```

Comment:
It is within the block `NS_ASSUME_NONNULL_BEGIN`, would this still be allowed? I thought this would give a compile or warning error. 
Otherwise I think we shouldn't allow for it to be `nil`; the correct way to cancel a gesture recognition is to use the delegate and `shouldRecognize*` return whether or not the gesture would be allowed.